spec/grid_spec.rb

```diff
@@ -4,36 +4,37 @@
 
   let(:grid) {Grid.new}
   let(:game) {Game.new(grid)}
+  let(:player) {Player.new(grid)}
 
   describe "registers positions of all ships placed" do
     it "registers submarine with its position in the list of ships placed" do
       submarine = Ship.new("submarine", 1)
 
-      grid.mark_ship_positions(5, "A", submarine, "vertical")
+      grid.mark_ship_positions(5, "A", submarine, "v")
```

Comment:
Oooh, here's the perfect example of why you should use symbols instead and convert the user input string into a symbol in the Ui. Then you wouldn't have to change the string "vertical" and "horizontal" in all the tests!